src/expr/src/scalar/func.rs

```diff
@@ -4827,6 +4837,9 @@ pub(crate) fn regexp_replace_parse_flags(flags: &str) -> (usize, Cow<'_, str>) {
     (limit, flags)
 }
 
+/// WARNING: This function has potential OOM risk if used with an inflationary
+/// replacement pattern. It is very difficult to calculate the output size ahead
+/// of time because the replacement pattern may depend on capture groups.
```

Comment:
Nit: Turn into a code comment, not a rustdoc comment, or document the function and add the warning.